spec/models/miq_report_spec.rb

```diff
@@ -115,6 +124,26 @@
 
       expect(report_result).to match_array(expected_results)
     end
+
+    let(:exp_3) { MiqExpression.new("IS NOT EMPTY" => {"field" => "#{vm_2.type}-#{virtual_column_key_3}"}) }
+
+    it "generates report with dynamic custom attributes with filtering with field which is not listed in cols" do
+      report.conditions = exp_3
+      report.queue_generate_table(:userid => user.userid)
+      report._async_generate_table(miq_task.id, :userid => user.userid, :mode => "async",
+                                   :report_source => "Requested by user")
+
+      report_result = report.table.data.map do |x|
+        x.data.delete("id")
+        x.data
+      end
+
+      expected_results = []
+      expected_results.push("name" => vm_2.name, virtual_column_key_1 => nil,
```

Comment:
Is there a reason to do `thing = []; thing.push(:foo => "bar")` over `thing = [{:foo => "bar"}]` here?